sharding/test/callbacks_test.go

```diff
@@ -71,12 +66,34 @@ func (t *CallbacksTestSuite) TestFailsRunOnPanicError() {
 	callbackReceived := false
 	t.ErrorCallback = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		callbackReceived = true
+
+		resp := t.requestMap(r)
+		errorData := make(map[string]string)
+		errorData["ErrFrom"] = "test_error"
+		errorData["ErrMessage"] = "test error"
+
+		errorDataBytes, jsonErr := json.MarshalIndent(errorData, "", "  ")
+		t.Require().Nil(jsonErr)
+		t.Require().Equal(string(errorDataBytes), resp["Payload"])
+
 		w.WriteHeader(http.StatusInternalServerError)
 	})
 
+	t.Ferry.Ferry.ErrorHandler = &ghostferry.PanicErrorHandler{
+		Ferry:         t.Ferry.Ferry,
+		ErrorCallback: t.Config.ErrorCallback,
+	}
+	panicReceived := false
+	defer func() {
+		if r := recover(); r != nil {
+			panicReceived = true
+			t.Require().Equal(r, "fatal error detected, see logs for details")
+		}
+	}()
 	t.Ferry.Ferry.ErrorHandler.Fatal("test_error", errors.New("test error"))
 
 	t.Require().True(callbackReceived)
+	t.Require().True(panicReceived)
```

Comment:
I don't think any of lines 93+ are executed. Try this example (https://play.golang.org/p/SPKXST8TKvw):

```
package main

import "fmt"

func main() {
    defer func() {
        if r := recover(); r != nil {
            fmt.Println("Recovered in main. Panic was:", r)
        }
    }()

    fmt.Println("Calling f.")

    f()

    fmt.Println("I'M NEVER EXECUTED!!!!")
}

func f() {
    panic("test")   
}
```

Reply:
True. I think I can fix this test if I implement the `ReportError` mechanism.